### PR TITLE
bug fixes for group

### PR DESCRIPTION
**Pull request by carchi8py**
##### SUMMARY
Bug Fixes for access_group

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME
- na_elementsw_access_group.py

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: lib/ansible/modules/storage/netapp/na_elementsw_access_group.py [hunk 1]
```diff
@@ -299,10 +303,9 @@ def apply(self):
...
         if self.account_id is not None:
-            self.account_id_valid = self.get_account_id()
-
+            self.account_id = self.get_account_id()
         if self.state == 'present' and self.volumes is not None:
-            if self.account_id_valid:
+            if self.account_id:
                 self.volumes = self.get_volume_id()
             else:
                 self.module.fail_json(msg='Error: Specified account id %s does not exist ' % self.account_id)
```
Comment (lonico) [inline lib/ansible/modules/storage/netapp/na_elementsw_access_group.py]: The error message is now useless (or will generate a trace back) as self.account_id is now None.

Maybe we could add
input_account_id = str(self.account_id) before line 304.
and change the error message to:
self.module.fail_json(msg='Error: Specified account id %s does not exist ' % input_account_id)

File: lib/ansible/modules/storage/netapp/na_elementsw_access_group.py [hunk 2]
```diff
@@ -320,10 +323,15 @@ def apply(self):
...
-                            changed = True
+                    if not group_detail.volumes:
+                        # If access group does not have any volume attached
+                        update_group = True
+                        changed = True
+                    else:
+                        for volumeID in group_detail.volumes:
+                            if volumeID not in self.volumes:
+                                update_group = True
+                                changed = True
```
Comment (lonico) [inline lib/ansible/modules/storage/netapp/na_elementsw_access_group.py:335]: a nitpick, we could break out of the loop as soon as we found a change

**Review by lonico — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/storage/netapp/na_elementsw_access_group.py (hunk 1 above)
Comment (thedoubl3j) [reply lib/ansible/modules/storage/netapp/na_elementsw_access_group.py]: @lonico this coming through the pipe or are we good with what is here?

**Review by lonico — APPROVED** [review]
shipit

**Review by thedoubl3j — APPROVED** [review]
bug fix and adding tests. lgtm. shipit

**Pull request merged**